src/completion.js

```diff
@@ -88,54 +119,62 @@ module.exports = class KiteCompletionProvider {
     }
 
-    const filename =  normalizeDriveLetter(document.fileName)
-    const filterText = buildFilterText(document, position)
+    const filename = normalizeDriveLetter(document.fileName);
+    const filterText = buildFilterText(document, position);
     // Use snippets completions.
-    if (workspace.getConfiguration('kite').enableSnippets) {
+    if (workspace.getConfiguration("kite").enableSnippets) {
       return this.getSnippetCompletions(document, text, filename, filterText);
     }
     // Use legacy completions.
     const cursorPosition = document.offsetAt(position);
     const payload = {
       text,
-      editor: 'vscode',
+      editor: "vscode",
       filename,
       cursor_runes: cursorPosition,
-      offset_encoding: 'utf-16',
+      offset_encoding: "utf-16"
     };
 
     Logger.debug(payload);
 
-    return this.Kite.request({
-      path: completionsPath(),
-      method: 'POST',
-    }, JSON.stringify(payload), document)
-    .then(data => {
-      data = parseJSON(data, {});
-      const completions = data.completions || [];
-      const length = String(completions.length).length;
-
-      return completions.map((c, i) => {
-        const item = new CompletionItem(c.display);
-        item.sortText = fill(String(i), length, '0');
-        item.insertText = c.insert;
-        // Use previous word, otherwise default to c.insert.
-        item.filterText = filterText ? filterText : c.insert;
-        if (c.documentation_text !== '') {
-          item.documentation = buildMarkdown(c.symbol.value[0].repr, c.hint, c.documentation_text);
-        }
-        // Note: The space following the Kite icon is the unicode space U+2003
-        // instead of the normal space U+0020 because VS Code strips the detail.
-        item.detail = c.hint + ' ⟠ ';
-        item.kind = kindForHint(c.hint);
-        return item;
-      });
-    })
-    .catch(() => []);
+    return this.Kite.request(
+      {
+        path: completionsPath(),
+        method: "POST"
+      },
+      JSON.stringify(payload),
+      document
+    )
+      .then(data => {
+        data = parseJSON(data, {});
+        const completions = data.completions || [];
+        const length = String(completions.length).length;
+
+        return completions.map((c, i) => {
+          const item = new CompletionItem(c.display);
+          item.sortText = fill(String(i), length, "0");
+          item.insertText = c.insert;
+          // Use previous word, otherwise default to c.insert.
+          item.filterText = filterText ? filterText : c.insert;
+          if (c.documentation_text !== "") {
+            item.documentation = buildMarkdown(
+              c.symbol.value[0].repr,
+              c.hint,
+              c.documentation_text
+            );
+          }
+          // Note: The space following the Kite icon is the unicode space U+2003
+          // instead of the normal space U+0020 because VS Code strips the detail.
+          item.detail = c.hint + " ⟠ ";
```

Comment:
do we also need to change this? why is this code duplicated?

Reply:
Updated. This is for the legacy completions codepath, so that's why it looks so similar.

Reply:
would it make sense to factor out values such as this into top-level constants?

Reply:
Yeah, absolutely. Will do before merging.